src/main/java/demon/Demon.java

```diff
@@ -0,0 +1,128 @@
+package demon;
+
+import java.nio.file.NoSuchFileException;
+
+import java.io.IOException;
+
+import java.util.Scanner;
+
+import java.time.format.DateTimeParseException;
+
+/**
+ * A chatbot that allows user to add, delete, and modify tasks or reminders.
+ */
+public class Demon {
+    private TaskList tasks;
+    private Ui ui;
+    String filePath = "src/main/taskList.txt";
+
+    /**
+     * Initializes the bot and load tasks saved in taskList.txt file
+     * if there is any previous tasks.
+     *
+     * @param filePath The relative path to the taskList.txt file
+     */
+    public Demon(String filePath) {
+        ui = new Ui();
+        final Storage STORAGE = new Storage(filePath);
```

Comment:
@Colex2000 You use `filePath` as both a parameter in your constructor (which you use in your `main` method in line 125, and as a instance variable (in line 17). I think this is an instance of variable shadowing which might lead to unforeseen bugs in the future. You might want to consider only using one instance of `filePath`.